compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirOverrideChecker.kt

```diff
@@ -130,6 +130,20 @@ object FirOverrideChecker : FirClassChecker() {
         }
     }
 
+    // See [DescriptorVisibilities.compare]
+    // If either one comes from Java, using the regular [Visibilities#compareTo], which uses enum ordinal, will return `null`, since there
+    // is no matched visibility. [JavaVisibilities] has Java-specific visibilities, along with their own [compareTo]. Hence, try comparison
+    // again with an opposite direction.
+    private fun compareVisibility(first: Visibility, second: Visibility): Int? {
```

Comment:
Can you just call `org.jetbrains.kotlin.descriptors.Visibilities#compare` instead of define the same logic here?

Reply:
We can't because we're not using `Descriptor*`.

Reply:
But the very `Visibility` class you are comparing is from the same package in the same module.

Reply:
Duh, I thought you said `DescriptorVisibitlies.compare` 🤦

Reply:
Thank you for catching this! Even much simpler now!